### PR TITLE
section commit for April 1st, '20

### PR DESCRIPTION
**Pull request by croqueGrec09**
includes commits for:
- ERMS-2260: subscription copy did not work because of pending change configuration transfer
- non-tested inheritance of auto-accepted pending changes

### CODE REVIEW

File: app/grails-app/domain/com/k_int/kbplus/PendingChange.groovy [hunk 1]
```diff
@@ -223,8 +249,17 @@ class PendingChange {
                 if(target instanceof IssueEntitlementCoverage) {
                     IssueEntitlementCoverage targetCov = (IssueEntitlementCoverage) target
                     targetCov[targetProperty] = parsedNewValue
-                    if(targetCov.save())
+                    if(targetCov.save()) {
+                        if(auditService.getAuditConfig(subscription,msgToken)) {
+                            subscription.derivedSubscriptions.each { Subscription childSub ->
+                                IssueEntitlementCoverage childCov = IssueEntitlementCoverage.executeQuery('select ic from IssueEntitlementCoverage ic where ic.issueEntitlement.tipp = :tipp and ic.issueEntitlement.subscription = :child',[child:childSub,tipp:targetCov.issueEntitlement.tipp])[0]
```
Comment (klodav) [inline app/grails-app/domain/com/k_int/kbplus/PendingChange.groovy:255]: [0] always exists?

File: app/grails-app/domain/com/k_int/kbplus/PendingChange.groovy [hunk 2]
```diff
@@ -245,8 +289,16 @@ class PendingChange {
             case PendingChangeConfiguration.COVERAGE_DELETED:
                 if(target instanceof IssueEntitlementCoverage) {
                     IssueEntitlementCoverage targetCov = (IssueEntitlementCoverage) target
-                    if(targetCov.delete())
+                    if(targetCov.delete()) {
+                        if(auditService.getAuditConfig(subscription,msgToken)) {
+                            subscription.derivedSubscriptions.each { Subscription childSub ->
+                                IssueEntitlementCoverage childCov = IssueEntitlementCoverage.executeQuery('select ic from IssueEntitlementCoverage ic where ic.issueEntitlement.tipp = :tipp and ic.issueEntitlement.subscription = :child',[child:childSub,tipp:targetCov.issueEntitlement.tipp])[0]
```
Comment (klodav) [inline app/grails-app/domain/com/k_int/kbplus/PendingChange.groovy:295]: [0] always exists?

**Comment by croqueGrec09** [comment]
Too early to deploy. Closed.

File: app/grails-app/controllers/com/k_int/kbplus/SubscriptionController.groovy [hunk 3]
```diff
@@ -5283,12 +5283,23 @@ class SubscriptionController extends AbstractDebugController {
                 if (params.subscription.copyPackages) {
                     baseSubscription.packages?.each { pkg ->
                         def pkgOapls = pkg.oapls
+                        Set<PendingChangeConfiguration> pcc = pkg.pendingChangeConfig
```
Comment (Moetez16) [inline app/grails-app/controllers/com/k_int/kbplus/SubscriptionController.groovy:5286]: Hast du auch an die anderen Stellen gedacht, die dies betreffen!

Comment (croqueGrec09) [reply app/grails-app/controllers/com/k_int/kbplus/SubscriptionController.groovy:5286]: Welche wären das?

Comment (Moetez16) [reply app/grails-app/controllers/com/k_int/kbplus/SubscriptionController.groovy:5286]: Du musst nach alle Stellen, wo def pkgOapls = pkg.oapls auftaucht suchen. 4 Stellen gibt es.

**Pull request closed without merging**